erpnext/public/js/utils.js

```diff
@@ -653,4 +653,108 @@ $(document).on('app_ready', function() {
 			});
 		});
 	}
+
+	$.each(["Purchase Receipt", "Stock Entry", "Delivery Note"], function(i, doctype) {
+		frappe.ui.form.on(doctype, {
+			refresh: function(frm) {
+				if(!frm.doc.is_return && frm.doc.status!="Closed") {
+					if (frm.doc.docstatus == 0) {
+						frm.add_custom_button(__('Make Quality Inspection'),
```

Comment:
Nest this under a "Make" dropdown

```suggestion
						frm.add_custom_button(__('Quality Inspections'),
```